src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLease.cs

```diff
@@ -72,6 +75,27 @@ public DateTime Timestamp
         [JsonProperty("_ts")]
         private long TS { get; set; }
 
+        [JsonProperty("LeaseToken", NullValueHandling = NullValueHandling.Ignore)]
+        private string LeaseToken
+        {
+            get
```

Comment:
Is this code really neeed? If we migrate from V3, the setter would guarantee to set ParititionId. That's all we need. Saving LeaseToken is not really needed, because v3 will do the same thing if it sees ParititionId, right?

The thing is that with splits, we will have partially filled leases anyhow, i.e. some will have both LeaseToken and PartitionId, thus there must be logic for leases that are not filled. If there has to be this logic, we can safely assume all leases are not filled. Maintaining both properties for some leases seems to be just extra complexity.

Reply:
Makes sense, removed